src/routes/search/+page.svelte

```diff
@@ -0,0 +1,95 @@
+<script lang="ts">
+	import Button from '$lib/components/Button.svelte'
+	import VideoList from '$lib/components/VideoList.svelte'
+	import { searchVideos } from '$lib/data'
+	import type { Video } from '$lib/data'
+
+	let videos: Video[]
+	$: videos = []
+
+	let searchQuery = ''
+
+	//TODO debounce (trailing edge) this function
+	const searchSubmit = () => {
+		videos = searchQuery === '' ? [] : searchVideos(searchQuery)
+	}
+</script>
+
+<div class="container">
+	<h1>Search</h1>
+
+	<form on:submit={searchSubmit}>
+		<label for="search" class="sr-only">Search Query</label>
+		<div id="field-wrapper">
+			<input
+				type="search"
+				id="search"
+				placeholder="search for something"
+				bind:value={searchQuery}
+			/>
+		</div>
+		<div id="button-wrapper">
+			<Button handler={() => {}}>Go Get It</Button>
```

Comment:
Minor thing, but `() => {}` could be set as a default value for this prop in the Button component so we don't have to include it when we don't need it.

Reply:
I was considering doing this, but then I ended up not doing it for ..whatever reason? I've changed my mind and made the change.